core/rest-client/src/http_client.rs

```diff
@@ -56,19 +55,76 @@ pub trait SendHttpRequest {
 		T: RestPath<U>;
 }
 
+/// Send trait used by the http client to send HTTP request, based on `http_req`.
+pub trait Send: Default {
+	fn execute_send_request(
+		&self,
+		request: &mut Request,
+		writer: &mut Vec<u8>,
+	) -> Result<Response, Error>;
+}
```

Comment:
Interesting solution - not what I had in mind when we discussed these improvements, but I'd say it's a good one nonetheless 👍